routers/repo/view.go

```diff
@@ -245,6 +245,25 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 
 // Home render repository home page
 func Home(ctx *context.Context) {
+	if len(ctx.Repo.Repository.Units) > 0 {
+		tp := ctx.Repo.Repository.Units[0].Type
+		if tp == models.UnitTypeCode {
+			renderCode(ctx)
+			return
+		}
+
+		unit, ok := models.Units[tp]
+		if ok {
+			ctx.Redirect(setting.AppSubURL + fmt.Sprintf("/%s%s",
+				ctx.Repo.Repository.FullName(), unit.URI))
+			return
+		}
+	}
+
+	ctx.Handle(404, "Home", fmt.Errorf(ctx.Tr("units.error.no_unit_allowed_repo")))
```

Comment:
Why are we leaking information that this repo exists but have nothing to show? IMO no valid Units should be the same as Private.